middleware.js

```diff
@@ -17,7 +17,7 @@ function logErrors (err, req, res, next) {
 
 
 function clientErrorHandler (err, req, res, next) {
-  if (req.xhr) {
+  if (req.xhr || req.headers["content-type"] === "application/json") {
```

Comment:
Sorry, I may have missed this earlier... but do we need to check `if (err && req.xhr || req.headers...)` here?
What's the likelihood that `err` is `undefined`, but either `req.xhr` or `req.headers["content-type"] === "application/json"` is truthy, which would cause `err.message` to be accessing an the ".message" property on an undefined object (`err`).

Reply:
Hmm ... I'm not sure. According to [Express error handling guide](https://expressjs.com/en/guide/error-handling.html) these handlers that are appended after all other middleware and routes will only be reached by thrown `Error`s, so in theory the `err` object will always exist.